File: code_eval/cpp/easy/challenge_93.cpp
// Created by cschalk on 3/21/16.
// Challenge 93 : Capitalize Strings
#include <iostream>
#include <fstream>
#include <locale>
#include <sstream>

using namespace std;

int main(int argc, char ** args) {
    std::ifstream input {args[1]};
    std::locale loc;
    std::string line;
    while (std::getline(input, line)) {
        std::string word;
        std::stringstream ss(line);
        while (ss >> word) {
            std::cout << std::toupper(word[0], loc);
            std::cout << word.substr(1, word.length() - 1);
            std::cout << " ";
        }
        std::cout << std::endl;
    }
    return 0;

}

File: code_eval/java/easy/Challenge93.java
package easy;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

/**
 * Challenge 93 : Capitalize Strings
 * Created by cschalk on 3/21/16.
 */
public class Challenge93 {
    public static void main (String[] args) throws Exception {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            capitalizeString(line);
        }
    }

    private static void capitalizeString(String input) {
        String[] words = input.split(" ");
        String output = "";
        for (String word : words) {
            output += capitalize(word).concat(" ");
        }
        System.out.println(output.trim());
    }

    private static String capitalize(String word) {
        String toReturn = ("" + word.charAt(0)).toUpperCase();
        return toReturn + word.substring(1);
    }
}
